### ISSUE TITLE
`Mutation` class isn't exposed from @tanstack/query-core

### ISSUE THREAD

**Issue by AiSirachcha21** [TanStack/query#7785]
### Describe the bug

I am attempting to create a small wrapper class around TanStack Query's Mutation API and I keep running into the following issue

```
WARNING in <redacted> 75:27-35
export 'Mutation' (imported as 'Mutation') was not found in '@tanstack/query-core' (possible exports: CancelledError, InfiniteQueryObserver, MutationCache, MutationObserver, QueriesObserver, Query, QueryCache, QueryClient, QueryObserver, defaultShouldDehydrateMutation, defaultShouldDehydrateQuery, dehydrate, focusManager, hashKey, hydrate, isCancelledError, isServer, keepPreviousData, matchMutation, matchQuery, notifyManager, onlineManager, replaceEqualDeep, skipToken)
```

The way I have imported it is as follows

```ts
import { Mutation } from "@tanstack/query-core"

export class MutationBuilder<TData = unknown> {
	mutation: Mutation;
	subscription: {
		unsubscribe: ReturnType<MutationObserver<TData>['subscribe']>;
	};

	constructor(
		private queryClient: QueryClient
	) { }

	build<T>(options: ConstructorParameters<typeof Mutation<T>>[0]["options"]): MutationBuilder<T> {
		const mutation = new Mutation({
			mutationCache: this.queryClient.getMutationCache(),
			mutationId: 1,
			options
		});

		options.mutationFn

		this.mutation = mutation;

		return this as any;
	}

	subscribe(subscribeFn: Parameters<MutationObserver["subscribe"]>[0], options?: MutationObserverOptions<TData>) {
		const mutationObserver = new MutationObserver<TData>(this.queryClient, options);
		this.mutation.addObserver(mutationObserver);

		this.subscription.unsubscribe = mutationObserver.subscribe(subscribeFn)

		return mutationObserver as Omit<MutationObserver<TData>, "subscribe" | "getCurrentResult">;
	}
}
```

I checked the source file at https://github.com/TanStack/query/blob/main/packages/query-core/src/mutation.ts but it seems to be exported correctly there.

### Your minimal, reproducible example

https://codesandbox.io/s/elated-firefly-vznnq4?file=/src/App.js

### Steps to reproduce

Run it in the browser, the problem will show up in the console stating `TypeError: _queryCore.Mutation is not a constructor`

### Expected behavior

I expected Mutation to be usable in order to setup Mutation queries alongside QueryClient

### How often does this bug happen?

Every time

### Platform

OS: MacOS
Browser: Arc (v 1.52.0), Chromium Engine Version 126.0.6478.183

### Tanstack Query adapter

vanilla

### TanStack Query version

5.49.1

### TypeScript version

4.9.4

**Comment by AiSirachcha21** [comment]
Will do

**Comment by AiSirachcha21** [comment]
@TkDodo do you think adding an example project for using the `query-core` package directly via a vanilla project might be useful in the project ? Thinking if I should add one to the examples directory ?

**Comment by lachlancollins** [comment]
Fixed in v5.51.15
